src/metabase/upload.clj

```diff
@@ -215,12 +216,25 @@
       (throw (ex-info (tru "Unable to recognise file separator")
                       {:status-code 422}))))
 
+(def ^:private allowed-extensions #{nil "csv" "tsv" "txt"})
+
+(def ^:private allowed-mime-types #{"text/csv" "text/tab-separated-values" "text/plain"})
+
+(def ^:private ^Tika tika (Tika.))
```

Comment:
is this fast?

Reply:
Haven't benchmarked it, but I'd be shocked if it's slower than the actual update. Are you worried about DoS?

Reply:
I guess we could swap the order with the auth check.

Reply:
nah, I'm worried about start up time being delayed a little bit more :) As for content check - I know it's not very fast, but I don't have any good ideas. Auth check should be done before content check for sure though.

Reply:
It's 1.5 ms, I think that's fine.